tests/MSBuildDeviceIntegration/Tests/InstallTests.cs

```diff
@@ -323,5 +323,35 @@ public void BlankAdbTarget ()
 				Assert.IsTrue (b.Output.IsTargetSkipped ("_BuildApkFastDev"), "_BuildApkFastDev should be skipped!");
 			}
 		}
+
+		static object [] PackageNamingPolicySource = new object [] {
+			new [] {
+				/* packageNamingPolicy */ "LowercaseMD5",
+				/* activityPackageName */ "md52d9cf6333b8e95e8683a477bc589eda5",
+			},
+			new [] {
+				/* packageNamingPolicy */ "LowercaseCrc64",
+				/* activityPackageName */ "crc64f75eeacfa0ca1368",
+			},
+		};
+
+		[Test]
+		[TestCaseSource (nameof (PackageNamingPolicySource))]
+		public void PackageNamingPolicy (string packageNamingPolicy, string activityPackageName)
+		{
+			if (!HasDevices)
+				Assert.Ignore ("Skipping Test. No devices available.");
+
+			var proj = new XamarinAndroidApplicationProject ();
+			proj.SetProperty ("AndroidPackageNamingPolicy", packageNamingPolicy);
+			proj.SetProperty (KnownProperties.AndroidSupportedAbis, "armeabi-v7a;x86");
+			using (var b = CreateApkBuilder ()) {
+				Assert.IsTrue (b.Install (proj), "Install should have succeeded.");
+				ClearAdbLogcat ();
+				AdbStartActivity ($"{proj.PackageName}/{activityPackageName}.MainActivity");
```

Comment:
This is why we should put `[Register]` on `MainActivity`, so that the name doesn't change based on `$(AndroidPackageNamingPolicy)`.